ci/build_scripts/build.sh

```diff
@@ -0,0 +1,174 @@
+#!/bin/bash -x
+
+set -euo pipefail
+
+help() {
+  cat <<EOF
+Compile and build the tedge components and debian packages.
+Cross is automatically used if you are trying to build for a foreign target (e.g. build for arm64 on a x86_64 machine)
+
+Usage:
+    $0 [ARCH] [--use-cross]
+
+Args:
+    ARCH     RUST target architecture which can be a value listed from the command 'rustc --print target-list'
+             If left blank then the TARGET will be set to that of the machine building the packages.
+
+    Example ARCH (target) values:
+
+        * x86_64-unknown-linux-gnu
+        * aarch64-unknown-linux-gnu
+        * aarch64-unknown-linux-musl
+        * arm-unknown-linux-gnueabihf
+        * armv7-unknown-linux-gnueabihf
+        * armv7-unknown-linux-musleabihf
+
+Flags:
+    --use-cross     Force to use cross to build the packages
+
+Examples:
+    $0
+    # Build for the current CPU architecture
+
+    $0 aarch64-unknown-linux-gnu
+    # Build for arm64 linux (gnu lib)
+
+    $0 x86_64-unknown-linux-gnu
+    # Build for x86_64 linux (gnu lib)
+
+    $0 armv7-unknown-linux-gnueabihf
+    # Build for armv7 (armhf) linux (gnu lib)
+
+    $0 --use-cross
+    # Force to use cross when building for the current architecture
+EOF
+}
+
+ARCH=
+TARGET=()
+
+REST_ARGS=()
+while [ $# -gt 0 ]
+do
+    case "$1" in
+        -c|--use-cross)
+            USE_CROSS=1
+            ;;
+
+        -h|--help)
+            help
+            exit 0
+            ;;
+        
+        *)
+            REST_ARGS+=("$1")
+            ;;
+    esac
+    shift
+done
+set -- "${REST_ARGS[@]}"
+
+if [ $# -eq 1 ]; then
+    ARCH="$1"
+fi
+
+# Install required cargo crates
+cargo install cargo-deb --version 1.38.1
+
```

Comment:
To use `cross` docker is required. So, I feel better to put it as a requirement and installing it.

Reply:
However the existing script was already expecting docker to be installed. And I don't think installing docker automatically is a good idea as the install highly depends on the OS so it would be too difficult to cover all different scenarios.

Currently we don't have any build/release documentation, so I think we should address this in another PR.

Reply:
Docker must not be installed implicitly.

Reply:
ok